netcode.c

```diff
@@ -3754,6 +3758,60 @@ void netcode_server_send_client_packet( struct netcode_server_t * server, void *
     server->client_last_packet_send_time[client_index] = server->time;
 }
 
+/**
+ * @brief skillz_print_all_matches
+ *
+ * Mostly just a test function.  Prints every match in the hash.
+ *
+ */
+void skillz_print_all_matches( struct netcode_server_t * server )
+{
+    skillz_match_t * match;
+
+    netcode_printf( NETCODE_LOG_LEVEL_INFO, "\n\nPrinting the matches and their clients.\n" );
+    for( match = server->skillz_matches; match != NULL; match = ( skillz_match_t * )( match->hh.next ) )
+    {
+        for( int i = 0; i < match->num_clients_in_match; ++i)
+        {
+            netcode_printf( NETCODE_LOG_LEVEL_INFO, "match id: %d client id: %d clients in match: %d\n",
+                    match->skillz_match_id, match->clients_in_match[i], match->num_clients_in_match );
+        }
+    }
+    netcode_printf( NETCODE_LOG_LEVEL_INFO, "\n\n" );
+}
+
+/**
+ * @brief skillz_match_disconnect
+ * @param server
+ * @param client_index
+ * @return bool if success.
+ *
+ * Removes the match from the hash table, frees the memeory.
+ */
+int skillz_match_disconnect( struct netcode_server_t * server, int client_index )
+{
+    int disconnect_match_id = server->skillz_match_id[client_index];
+
+    skillz_match_t * match;
+    HASH_FIND_INT( server->skillz_matches, &disconnect_match_id, match );
+
+    if( !match )
+    {
+        netcode_printf(NETCODE_LOG_LEVEL_INFO, "match %d did not exist\n",
+                       disconnect_match_id);
+        return 0;
+    }
+
+    HASH_DEL( server->skillz_matches, match );
```

Comment:
Does this also remove the client(s)?

Reply:
The client who has not sent a disconnect token has not been disconnected.  This is because there is a 'TTL' on a match, preventing the auto-kicking of clients.  They will be kicked from the server, but that is for another story.  Should I add that they get kicked right now?

Reply:
Nope, just checking